From 05938a167dd4190014d800bb02446e2779a9b3b2 Mon Sep 17 00:00:00 2001
From: Elvis Angelaccio <elvis.angelaccio@kde.org>
Date: Thu, 16 Nov 2017 12:06:41 +0100
Subject: Fix testtrash with qtbase 5.10 beta 4

Summary:
Same fix as in D8836, this time for testtrash.

Test Plan:
Run testtrash.

Reviewed By: dfaure

Differential Revision: https://phabricator.kde.org/D8837

*asturm: Fix build with Qt-5.7.1
---
 src/core/listjob.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/core/listjob.cpp b/src/core/listjob.cpp
index 4b7b510..54a422a 100644
--- a/src/core/listjob.cpp
+++ b/src/core/listjob.cpp
@@ -117,7 +117,11 @@ void ListJobPrivate::slotListEntries(const KIO::UDSEntryList &list)
                 itemURL = q->url();
                 filename = entry.stringValue(KIO::UDSEntry::UDS_NAME);
                 Q_ASSERT(!filename.isEmpty()); // we'll recurse forever otherwise :)
-                itemURL.setPath(itemURL.path() + '/' + filename);
+                if (itemURL.path() == QLatin1String("/")) {
+                    itemURL.setPath(itemURL.path() + filename);
+                } else {
+                    itemURL.setPath(itemURL.path() + '/' + filename);
+                }
             }
 
             if (entry.isDir() && !entry.isLink()) {
-- 
cgit v0.11.2

